build-system/tasks/visual-diff/dev-mode.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+const inquirer = require('inquirer');
+const path = require('path');
+const puppeteer = require('puppeteer'); // eslint-disable-line no-unused-vars
+const {
+  verifySelectorsInvisible,
+  verifySelectorsVisible,
+  waitForPageLoad,
+} = require('./verifiers');
+const {cyan, yellow} = require('kleur/colors');
+const {HOST, PORT} = require('./consts');
+const {log} = require('./log');
+const {newPage} = require('./browser');
+const {sleep} = require('./helpers');
+const {WebpageDef} = require('./types');
+
+const ROOT_DIR = path.resolve(__dirname, '../../../');
+
+/**
+ * Runs a development mode.
+ *
+ * @param {!puppeteer.Browser} browser a Puppeteer controlled browser.
+ * @param {!Array<!WebpageDef>} webpages an array of JSON objects containing
+ *     details about the webpages to snapshot.
+ * @return {Promise<void>}
+ */
+async function devMode(browser, webpages) {
+  /** @type {WebpageDef} */
+  let webpage;
+  if (webpages.length > 1) {
+    webpage = (
+      await inquirer.prompt([
+        {
+          type: 'list',
+          name: 'webpage',
+          message:
+            'Select test name from ' +
+            cyan('visual-diff.jsonc') +
+            ' (use ' +
+            cyan('--grep') +
+            ' to filter this list):',
+          choices: webpages.map((webpage) => ({
+            name: webpage.name,
+            value: webpage,
+          })),
+        },
+      ])
+    ).webpage;
+  } else {
+    webpage = webpages[0];
+    log(
+      'info',
+      'Using test',
+      yellow(webpage.name),
+      '(Only one test matches the',
+      cyan('--grep'),
+      'value)'
+    );
+  }
```

Comment:
Breaking this out into a helper function could help with the readability and allow `webpage` to be `const` and always defined.

Reply:
Good idea! Done.